src/XRP/issued-currency-client.ts

```diff
@@ -0,0 +1,88 @@
+import { Wallet, XrplNetwork } from 'xpring-common-js'
+import GrpcNetworkClient from './network-clients/grpc-xrp-network-client'
+import GrpcNetworkClientWeb from './network-clients/grpc-xrp-network-client.web'
+import { XrpNetworkClient } from './network-clients/xrp-network-client'
+import isNode from '../Common/utils'
+import CoreXrplClient from './core-xrpl-client'
+import { AccountSetFlag } from './shared/account-set-flag'
+import { SetFlag } from './Generated/web/org/xrpl/rpc/v1/common_pb'
+import { AccountSet } from './Generated/web/org/xrpl/rpc/v1/transaction_pb'
+import TransactionResult from './shared/transaction-result'
+
+/**
+ * IssuedCurrencyClient is a client for working with Issued Currencies on the XRPL.
+ * @see https://xrpl.org/issued-currencies-overview.html
+ */
+export default class IssuedCurrencyClient {
+  private coreXrplClient: CoreXrplClient
+
+  /**
+   * Create a new IssuedCurrencyClient.
+   *
+   * The IssuedCurrencyClient will use gRPC to communicate with the given endpoint.
+   *
+   * @param grpcUrl The URL of the gRPC instance to connect to.
+   * @param network The network this IssuedCurrencyClient is connecting to.
+   * @param forceWeb If `true`, then we will use the gRPC-Web client even when on Node. Defaults to false. This is mainly for testing and in the future will be removed when we have browser testing.
+   */
+  public static issuedCurrencyClientWithEndpoint(
+    grpcUrl: string,
+    network: XrplNetwork,
+    forceWeb = false,
+  ): IssuedCurrencyClient {
+    return isNode() && !forceWeb
+      ? new IssuedCurrencyClient(new GrpcNetworkClient(grpcUrl), network)
+      : new IssuedCurrencyClient(new GrpcNetworkClientWeb(grpcUrl), network)
+  }
+
+  /**
+   * Create a new IssuedCurrencyClient with a custom network client implementation.
+   *
+   * In general, clients should prefer to call `issuedCurrencyClientWithEndpoint`. This constructor is provided to improve testability of this class.
+   *
+   * @param networkClient A network client which will manage remote RPCs to Rippled.
+   * @param network The network this IssuedCurrencyClient is connecting to.
+   */
+  public constructor(
+    private readonly networkClient: XrpNetworkClient,
```

Comment:
```suggestion
    networkClient: XrpNetworkClient,
```

Reply:
I realized that, at least for the moment, we don't actually need direct access to the gRPC network client in this class - we just need it to feed to the `CoreXrplClient` constructor.. so by removing it as an instance variable we can get rid of everything related to shutUpCompiler.

Reply:
oh lit